libcontainer/nsenter/nsexec.c

```diff
@@ -570,20 +571,46 @@ void nsexec(void)
 							bail("failed to sync with child: write(SYNC_RECVPID_ACK)");
 						}
 					}
-
-					ready++;
-					break;
-				case SYNC_RECVPID_ACK:
-					/* We should _never_ receive acks. */
-					kill(child, SIGKILL);
-					bail("failed to sync with child: unexpected SYNC_RECVPID_ACK");
 					break;
 				case SYNC_CHILD_READY:
-					ready++;
+					ready = true;
 					break;
 				default:
-					bail("unexpected sync value");
+					bail("unexpected sync value: %u", s);
```

Comment:
While I understand this makes the code nicer, remember that the _only_ debugging information we normally get is the return code. Maybe we don't lose too much information, but something to keep in mind.